src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,95 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { debounce } from 'lodash';
+import { ChangeEvent, useCallback, useEffect, useRef, useState } from 'react';
+import { FilterType } from '../../types/Filter';
+
+type Props = {
+  setQuery: (value: string) => void;
+  query: string;
+  handleChangeSelect: (value: FilterType) => void;
+  filter: FilterType;
+  setFilter: (value: FilterType) => void;
+};
+export const TodoFilter: React.FC<Props> = ({
+  setQuery,
+  query,
+  handleChangeSelect,
+  filter,
+  setFilter,
+}) => {
+  const [inputValue, setInputValue] = useState(query);
+
+  const debouncedSetQuery = useCallback(
+    debounce((value: string) => {
+      setQuery(value);
+    }, 500),
+    [],
```

Comment:
The dependency array for `useCallback` should include `setQuery` to ensure that `debouncedSetQuery` is updated if `setQuery` changes. This is important to avoid potential stale closures.